### ISSUE TITLE
inference_detector result and show_result_meshlab _pred.obj unmatching columns (x and y swapped) CENTERPOINT

### ISSUE THREAD

**Issue by aldakata** [open-mmlab/mmdetection3d#1038]
**Describe the bug**
inference_detector result has a LiDARInstance3DBoxes where each row should be (x, y, z, x_size, y_size, z_size, yaw, ...)
But x and y appear to be swapped (after calculating corners and x_size and y_size)

**Reproduction**
1. What command or script did you run?
I run this code:
```python
from mmdet3d.apis import init_model, inference_detector, show_result_meshlab
import numpy as np
import open3d as o3d

pointcloud = "/midday/2021-09-20-14-32-46_12/6087_401541630.bin" # My pointcloud, format has to be NuScenes (i.e. each row needs 5 values (x,y,z,intensity,ring?))
CONFIG = './configs/centerpoint/centerpoint_01voxel_second_secfpn_circlenms_4x8_cyclic_20e_nus.py'
CHECKPOINT = './checkpoints/centerpoint_01voxel_second_secfpn_circlenms_4x8_cyclic_20e_nus_20201001_135205-5db91e00.pth' # model zoo downloadable .pth

model = init_model(CONFIG, CHECKPOINT, device='cuda:0')
result, data = inference_detector(model, pointcloud)
show_result_meshlab(data, result, './demo', 0, show=True, snapshot=False, task='det')
```
When running show_result_meshlab it generates 
./demo/6087_401541630/6087_401541630_points.obj
./demo/6087_401541630/6087_401541630_pred.obj

I looked at the file /6087_401541630_pred.obj and its first 8 rows (I understand that there is a row for each BBOX corner) were:
```python
v -34.71527345 21.18418483 -2.22205073
v -34.71527345 21.18418483 -0.59199709
v -32.77855068 21.13697963 -2.22205073
v -32.77855068 21.13697963 -0.59199709
v -34.82494206 16.68472905 -2.22205073
v -34.82494206 16.68472905 -0.59199709
v -32.88821928 16.63752385 -2.22205073
v -32.88821928 16.63752385 -0.59199709
```

But if I print the result corners (the python variable from my code above) I get:
```python
>>> result[0]['pts_bbox']['boxes_3d'].corners.tolist()[0]
#CORNERS
[[16.684728622436523, 34.82494354248047, -2.222050666809082],
 [16.684728622436523, 34.82494354248047, -0.5919970273971558],
 [21.184185028076172, 34.715274810791016, -0.5919970273971558],
 [21.184185028076172, 34.715274810791016, -2.222050666809082],
 [16.637523651123047, 32.88821792602539, -2.222050666809082],
 [16.637523651123047, 32.88821792602539, -0.5919970273971558],
 [21.136980056762695, 32.77854919433594, -0.5919970273971558],
 [21.136980056762695, 32.77854919433594, -2.222050666809082]]
```
If you look close it appears that the x column and the y column are swapped, either in the 6087_401541630_pred.obj or in the inference_detector result.

I checked the x_size, y_size and z_size to see if those were swapped too:
```python
>>> result[0]['pts_bbox']['boxes_3d'].tensor[0]
#output is (x, y, z, x_size, y_size, z_size, yaw, ...) according to LiDARInstance3DBoxes https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3d/core/bbox/structures/lidar_box3d.py 
tensor([18.9109, 33.8017, -2.2221,  1.9373,  4.5008,  1.6301,  1.5952, -3.3893,
         0.2190])
```
so  I should have:
x_size 1.9373
y_size 4.5008
z_size 1.6301

but I manually calculate the sizes, like this, choosing two values for each dimension from #CORNERS (above) where only that dimension changes.
```python
xdist = 21.184185028076172 - 16.684728622436523
ydist = 34.82494354248047 - 32.88821792602539
zdist = -2.222050666809082 + 0.5919970273971558
xdist, ydist, zdist
#And I get
(4.499456405639648, 1.9367256164550781, -1.6300536394119263)
```

So either (x, y, z, x_size, y_size, z_size, yaw, ...) is really (y, x, z, x_size, y_size, z_size, yaw, ...) which would make sense with 6087_401541630_pred.obj
or it is (x, y, z, y_size, x_size, z_size, yaw, ...) which would mean that show_result_meshlab swaps x column for y column (or else...)

2. Did you make any modifications on the code or config? Did you understand what you have modified?
I have not modified the config nor the code.

3. What dataset did you use?
I only used Inference.

**Environment**

1. Please run `python mmdet3d/utils/collect_env.py` to collect necessary environment infomation and paste it here.

```
A placeholder for trackback.
```

**Comment by ZCMax** [comment]
Actually the bboxes in results are in LiDAR coordinates, to show in MeshLab, we need first convert them to Depth Coordinates. In v1.0.0.dev0 branch, we follow [the coordinates settings](https://mmdetection3d--1001.org.readthedocs.build/en/1001/tutorials/coord_sys_tutorial.html).

**Comment by aldakata** [comment]
@ZCMax thank you very much for your answer, I now understand the transformation from _pref.obj and the inference detector x,y,z result.

But this doesn't explain the mismatch with the inference_detector result x_size, y_size, z_size.

The result is:
```python
tensor([18.9109, 33.8017, -2.2221,  1.9373,  4.5008,  1.6301,  1.5952, -3.3893, 0.2190])
```
According to the code it is (x, y, z, x_size, y_size, z_size, yaw, ...)

If I get the corners of the box (which are in LiDAR coordinates):
```python
>>> result[0]['pts_bbox']['boxes_3d'].corners.tolist()[0]
#CORNERS
[[16.684728622436523, 34.82494354248047, -2.222050666809082],
 [16.684728622436523, 34.82494354248047, -0.5919970273971558],
 [21.184185028076172, 34.715274810791016, -0.5919970273971558],
 [21.184185028076172, 34.715274810791016, -2.222050666809082],
 [16.637523651123047, 32.88821792602539, -2.222050666809082],
 [16.637523651123047, 32.88821792602539, -0.5919970273971558],
 [21.136980056762695, 32.77854919433594, -0.5919970273971558],
 [21.136980056762695, 32.77854919433594, -2.222050666809082]]

corner1 = (16.684728622436523, 34.82494354248047, -0.5919970273971558)
corner2 = (21.184185028076172, 34.715274810791016, -0.5919970273971558)
x_difference = 21.184185028076172 - 16.684728622436523 #RESULT 4.499456405639648
```
corner1 and corner2 share y and z axes, so, the difference between its x component (aprox. 4.45) will be the x_size of the box but according to this (x, y, z, x_size, y_size, z_size, yaw, ...) the x_size is 1.9373 and the y_size 4.5008.

Am I missing something?

**Comment by ZCMax** [comment]
I guess you must using the code of `master` branch, we did a coordinates refactor from `master` branch to `v1.0.0.dev0` branch. For LiDAR coordinates, the yaw definition has changed:

 **In `master` branch:** `The yaw is 0 at the negative direction of y axis, and decreases from
    the negative direction of y to the positive direction of x.`

**In `v1.0.0.dev0` branch:** `The yaw is 0 at the positive direction of x axis.`

We indeed define box represention as: `(x, y, z, x_size, y_size, z_size, yaw, ...)`. According to our detailed definition in [the coordinates settings](https://mmdetection3d--1001.org.readthedocs.build/en/1001/tutorials/coord_sys_tutorial.html), due the change of yaw definition, the difference is:

 **In `master` branch:** `(x, y, z, x_size, y_size, z_size, yaw, ...)` == `(x, y, z, w, l, h, yaw1, ...)`
**In `v1.0.0.dev0` branch:** `(x, y, z, x_size, y_size, z_size, yaw, ...)` == `(x, y, z, l, w, h, yaw2, ...)`

Then we go back to your case, since you are following the master coordinate sys:

The result is:
```python
tensor([18.9109, 33.8017, -2.2221,  1.9373,  4.5008,  1.6301,  1.5952, -3.3893, 0.2190])
```
It means,  for this object, `w=1.9373, l=4.5008, h=1.6301, yaw1=1.5952`, it shows the object is heading almost along the x-axis. So your calculated `x_difference` almost equals to object's `length (l)`, as known as `y_size`.
